lib/common/memory/src/mmap_type.rs

```diff
@@ -389,6 +389,38 @@ impl MmapBitSlice {
 
         Ok(())
     }
+
+    /// Reopens the mmap file with a larger size.
+    ///
+    /// # Arguments
+    /// - `path`: Path to the mmap file
+    /// - `total_capacity`: New total capacity of the bitslice, in amount of bits it should fit.
+    ///
+    /// # Safety
+    /// Marked unsafe because the path must be the same as the original mmap file.
+    pub unsafe fn extend(
```

Comment:
I reacal we have similar functionality somewhere, which didn't require unsafe function

cc @xzfc

Reply:
https://github.com/qdrant/qdrant/blob/1beab3d8aeea53886816ba82615fa3ef5222af3d/lib/segment/src/vector_storage/dense/dynamic_mmap_flags.rs#L163-L170

Reply:
TIL about `DynamicMmapFlags`, thanks @xzfc!